### PR TITLE
[Doc] Replace tensor by field for files beginning with t to z

### PR DESCRIPTION
**Pull request by Rullec**
Related issue = #1500 

This PR tries to replace `tensor` by `feld` for files beginning with `t` to `z`
- [x] taichicon.rst
- [x] tensor_matrix.rst
- [x] type.rst
- [x] utilities.rst
- [x] vector.rst
- [x] versioning_releases.rst

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----

### CODE REVIEW

**Comment by Rullec** [comment]
Hi @archibate !

So I have mostly updated all tensors to fields in our documents.

And the changes are broken into 3 seperate PRs: 1576, 1577, 1579. Would you please give me some suggestions?

File: docs/tensor_matrix.rst [hunk 1]
```diff
@@ -1,41 +1,41 @@
 .. _tensor:
 
-Tensors and matrices
-====================
+Fields and matrices
+===================
```
Comment (archibate) [inline docs/tensor_matrix.rst:4]:
```suggestion
Matrix fields
=============
```
The same to `Vector fields`, it should sounds more than intuitive for STEM students, WDYT? @yuanming-hu @Rullec

File: docs/tensor_matrix.rst [hunk 2]
```diff
@@ -1,41 +1,41 @@
...
-    Although mathematically matrices are treated as 2D tensors, in Taichi, **tensor** and **matrix** are two completely different concepts.
-    Matrices can be used as tensor elements, so you can have tensors with each element being a matrix.
+    Although mathematically matrices are treated as 2D fields, in Taichi, **field** and **matrix** are two completely different concepts.
+    Matrices can be used as field elements, so you can have fields with each element being a matrix.
 
-Tensors of scalars
-------------------
-* Every global variable is an N-dimensional tensor.
+Fields of scalars
+-----------------
```
Comment (archibate) [inline docs/tensor_matrix.rst:15]:
```suggestion
Scalar fields
-------------
```

File: docs/tensor_matrix.rst [hunk 3]
```diff
@@ -1,41 +1,41 @@
...
+* Every global variable is an N-dimensional field.
 
-  - Global ``scalars`` are treated as 0-D tensors of scalars.
+  - Global ``scalars`` are treated as 0-D fields of scalars.
```
Comment (archibate) [inline docs/tensor_matrix.rst:18]:
```suggestion
  - Global ``scalars`` are treated as 0-D scalar fields.
```
Please update the following changes too, I don't have time to comment them one-by-one.

File: docs/tensor_matrix.rst (hunk 1 above)
Comment (Rullec) [reply docs/tensor_matrix.rst:4]: It's a good idea. But I guess the main task of this PR is simply do replacement, and make sure the replacements have no error.
I would personally prefer to refine the whole documents, make them more fluent and accurate in another PR, HDYT?

Comment (archibate) [reply docs/tensor_matrix.rst:4]: Yes, thank for the reminder, let's do `matrix fields` iapr.

**Pull request merged**